opm/autodiff/SimulatorFullyImplicitBlackoilOutput.hpp

```diff
@@ -581,43 +603,163 @@ namespace Opm
+                rstKeywords["RVSAT"] = 0;
+                output.emplace_back(data::CellData{
                         "RVSAT",
                         Opm::UnitSystem::measure::oil_gas_ratio,
-                        std::move(adbToDoubleVector(sd.rvSat))});
+                        std::move(adbToDoubleVector(sd.rvSat)),
+                        true});
             }
 
 
             /**
              * Bubble point and dew point pressures
              */
-            if (vapour_active && liquid_active && outKeywords["PBPD"] > 0) {
-                outKeywords["PBPD"] = 0;
+            if (vapour_active && liquid_active && rstKeywords["PBPD"] > 0) {
+                rstKeywords["PBPD"] = 0;
                 Opm::OpmLog::warning("Bubble/dew point pressure output unsupported",
                         "Writing bubble points and dew points (PBPD) to file is unsupported, "
                         "as the simulator does not use these internally.");
             }
 
             //Warn for any unhandled keyword
-            for (auto& keyValue : outKeywords) {
+            for (auto& keyValue : rstKeywords) {
                 if (keyValue.second > 0) {
                     std::string logstring = "Keyword '";
                     logstring.append(keyValue.first);
                     logstring.append("' is unhandled for output to file.");
                     Opm::OpmLog::warning("Unhandled output keyword", logstring);
                 }
             }
+        }
+
+
+
+
+        /**
+         * Checks if the summaryConfig has a keyword with the standardized field, region, or block prefixes.
+         */
+        inline bool hasFRBKeyword(const SummaryConfig& summaryConfig, const std::string keyword) {
+            std::string field_kw = "F" + keyword;
+            std::string region_kw = "R" + keyword;
+            std::string block_kw = "B" + keyword;
+            return summaryConfig.hasKeyword(field_kw)
+                    || summaryConfig.hasKeyword(region_kw)
+                    || summaryConfig.hasKeyword(block_kw);
+        }
+
+
+        /**
+         * Returns the data as asked for in the summaryConfig
+         */
+        template<class Model>
+        void getSummaryData(
+                std::vector<data::CellData>& output,
+                const Opm::PhaseUsage& phaseUsage,
+                const Model& physicalModel,
+                const SummaryConfig& summaryConfig) {
+
+            typedef Opm::AutoDiffBlock<double> ADB;
+
+            const typename Model::SimulatorData& sd = physicalModel.getSimulatorData();
+
+            //Get shorthands for water, oil, gas
+            const int aqua_active = phaseUsage.phase_used[Opm::PhaseUsage::Aqua];
+            const int liquid_active = phaseUsage.phase_used[Opm::PhaseUsage::Liquid];
+            const int vapour_active = phaseUsage.phase_used[Opm::PhaseUsage::Vapour];
 
-            return simProps;
+            /**
+             * Now process all of the summary config files
```

Comment:
> based on rough and wildly inaccurate guessing

I'll join that game...

The checking should be O(1), while the copying is O(n) for n cells. So while it may be insignificant for small cases, it would consume significant time and memory on large ones.